inlang/source-code/sdk/src/persistence/slotfiles/createSlotWriter.test.ts

```diff
@@ -0,0 +1,145 @@
+		const slotStorage = await createSlotWriter<DocumentExample>({
+			fs: memoryFs,
+			path: "/slot/",
+			fileNameCharacters: 3,
+			slotsPerFile: 16,
+			watch: false,
+		})
+		const nonExistingRecord: DocumentExample = {
+			id: "1",
+			content: "first document",
+		}
+		try {
+			await slotStorage.update(nonExistingRecord)
+		} catch (e) {
+			expect((e as Error).message).contain("does not exist")
+		}
+	})
+
+	it("insert should lead to a transient record save should lead to a record in non transient state", async () => {
+		const memoryFs = createNodeishMemoryFs()
+		const slotStorage = await createSlotWriter<DocumentExample>({
+			fs: memoryFs,
+			path: "/slot/",
+			fileNameCharacters: 3,
+			slotsPerFile: 16,
+			watch: false,
+		})
+
+		const insertedDocument: DocumentExample = {
+			id: "1",
+			content: "first document",
+		}
+
+		await slotStorage.insert(insertedDocument, false)
+
+		expect(slotStorage._writerInternals.transientSlotEntries.size).eq(1)
+
+		await slotStorage.save()
+		expect(slotStorage._writerInternals.transientSlotEntries.size).eq(0)
+		const docs1 = await slotStorage.findDocumentsById([insertedDocument.id])
+		expect(docs1.length).eq(1)
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion -- we check the length before
+		expect(docs1[0]!.data.content).eq(insertedDocument.content)
+	})
+
+	it("simultanous updates should lead to a local conflict", async () => {
+		const memoryFs = createNodeishMemoryFs()
+		const slotStorage1 = await createSlotWriter<DocumentExample>({
+			fs: memoryFs,
+			path: "/slot/",
+			fileNameCharacters: 3,
+			slotsPerFile: 16,
+			watch: false,
+		})
+
+		const slotStorage2 = await createSlotWriter<DocumentExample>({
+			fs: memoryFs,
+			path: "/slot/",
+			fileNameCharacters: 3,
+			slotsPerFile: 16,
+			watch: false,
+		})
+
+		const insertedDocument: DocumentExample = {
+			id: "1",
+			content: "first document",
+		}
+		await slotStorage1.insert(insertedDocument, false)
+		expect(slotStorage1._writerInternals.transientSlotEntries.size).eq(1)
+		await slotStorage1.save()
+		expect(slotStorage1._writerInternals.transientSlotEntries.size).eq(0)
+
+		// force reload to avoid waiting for fs event
+		await slotStorage2.loadSlotFilesFromWorkingCopy(true)
+
+		const currentDocumentsStorage1 = await slotStorage1.readAll()
+		const currentDocumentsStorage2 = await slotStorage2.readAll()
+
+		expect(currentDocumentsStorage1).deep.eq(currentDocumentsStorage2)
+		// slotStorage2.disconnect()
+
+		const updateInSlot1 = structuredClone(insertedDocument)
+		updateInSlot1.content = "updated in slot1"
+		await slotStorage1.update(updateInSlot1, false)
+
+		const updateInSlot2 = structuredClone(insertedDocument)
+		updateInSlot2.content = "updated in slot2"
+		await slotStorage2.update(updateInSlot2, false)
+
+		await slotStorage1.save()
+		await slotStorage2.save()
+		const conflictingRecord = slotStorage2.findDocumentsById([insertedDocument.id])[0]
```

Comment:
isnt locking supposed to prevent this case? we need locking in any case to now destroy the file write on concurrent threads, so it should be simple to just prevent any writes from other threads concurrently.